draft-ietf-quic-load-balancers.md

```diff
@@ -687,23 +687,46 @@ packets that don't include a valid token.
 Both server and service must have access to Universal time, though tight
 synchronization is unnecessary.
 
+The tokens are protected using AES128-GCM AEAD, as explained in
+{{token-protection-with-aead}}.
 All tokens, generated by either the server or retry service, MUST use the
-following format. This format is the cleartext version. On the wire, these
-fields are encrypted using an AES-ECB cipher and the token key. If the token is
-not a multiple of 16 octets, the last block is padded with zeroes.
-
+following format, which includes:
+
+- The client IP address, serialized and used as associated data when
+  computing the AEAD encryption.
+- A token sequence number that must be used only once with a given
+  key. The token sequence number is transmitted in clear text, but
+  is protected as part of the AEAD associated data.
+- A token body, encoding the Original Destination Connection ID,
+  the Retry Source Connection ID, and the Timestamp, optionally
+  followed by server specific Opaque Data.
+
+DISCUSS: shall we specify how the token sequence is formatted? There
+are two main possibilities, a large random number that is statistically
+unique, or the combination of a server ID and a sequence number unique
+to hat server.
+
+DISCUSS: some implementations of QUIC encode in the token the Initial Packet
+Number used by the client, in order to verify that the client sends the
```

Comment:
Can you file a separate issue for that? It applies to both types of Retry Service and we should address it consistently for both.

I don't feel strongly one way or the other. I believe the point of that check is so that *some* servers enforce compliance to encourage good behavior, and I don't think Retry Services need to be part of that effort.

Reply:
The PN s only used with the RETRY TOKEN format, which is server specific. So it fits naturally in the server specific part.

Reply:
Yes, but a non-shared-state retry service might be required to include the initial packet number so that the server can do the validation.

That said, making the retry service do initial header decryption is a terrible idea. So let's rule out having any type of retry service encode the packet number.